packages/nimble-components/src/table/types.ts

```diff
@@ -24,3 +24,15 @@ export type TableFieldValue =
 export interface TableRecord {
     [key: TableFieldName]: TableFieldValue;
 }
+
+export type StringField<FieldName extends string> = {
+    [name in FieldName]: string | null | undefined;
+};
+
+export interface TableCellState<
+    TCellData extends TableRecord,
```

Comment:
I don't want to re-hash all our naming discussions from the end of last year, but similar to one of my other comments, I'm wondering if this should be `TCellRecord` to better convey that it is the user's data that is being used to create a cell.

Reply:
Updated.